emergence_lib/src/graphics/ui/hover_panel.rs

```diff
@@ -0,0 +1,180 @@
+                    TextSection::new("Position: ", key_text_style.clone()),
+                    TextSection::from_style(value_text_style.clone()),
+                ]),
+                PositionText,
+            ));
+
+            // Organism type
+            parent.spawn((
+                TextBundle {
+                    text: Text::from_sections([
+                        TextSection::new("Organism: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                    ]),
+                    visibility: Visibility::INVISIBLE,
+                    ..default()
+                },
+                OrganismText,
+            ));
+
+            // Crafting stuff
+            parent.spawn((
+                TextBundle {
+                    text: Text::from_sections([
+                        TextSection::new("Inputs: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                        TextSection::new("\nOutputs: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                        TextSection::new("\nActive recipe: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                        TextSection::new("\nStatus: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                    ]),
+                    visibility: Visibility::INVISIBLE,
+                    ..default()
+                },
+                CraftingText,
+            ));
+        })
+        .id();
+
+    commands.entity(right_panel).add_child(hover_panel);
+}
+
+/// Update the information displayed in the hover panel.
+pub fn update_hover_panel(
+    cursor_tile_pos: Res<CursorTilePos>,
+    hover_details: Res<HoverDetails>,
+    mut panel_query: Query<&mut Visibility, With<HoverPanel>>,
+    mut position_query: Query<&mut Text, With<PositionText>>,
+    mut organism_query: Query<
+        (&mut Text, &mut Visibility),
+        (
+            With<OrganismText>,
+            // Avoid conflicting queries
+            Without<PositionText>,
+            Without<HoverPanel>,
+        ),
+    >,
+    mut crafting_query: Query<
+        (&mut Text, &mut Visibility),
+        (
+            // Avoid conflicting queries
+            With<CraftingText>,
+            Without<PositionText>,
+            Without<HoverPanel>,
+            Without<OrganismText>,
+        ),
+    >,
+) {
+    if let Some(cursor_tile_pos) = cursor_tile_pos.0 {
+        // Update visibility of the whole panel
+        *panel_query.single_mut() = Visibility::VISIBLE;
+
+        // Update position text
+        position_query.single_mut().sections[1].value =
+            format!("{}, {}", cursor_tile_pos.x, cursor_tile_pos.y);
+
+        // Update organism text
+        if let Some(organism_details) = &hover_details.0 {
```

Comment:
These .0 calls always pain me: it's really hard to figure out the intent without going to the definition.

Reply:
How would you suggest to improve this?
Since the components are often just wrappers over more or less the same thing, the accessor functions would probably need the same name as the component itself, so the name would essentially be duplicated.
Is there a better way?

Reply:
Alright I tried that, unfortunately it needs `&**`, two derefs to get to the inner one and then the ref to not get a burrowing problem.
Not sure if this is more readable...

Reply:
I'm using reader methods instead, I think the `&**` stuff is a lot more confusing than the `.0`.